### DISCUSSION TITLE
Proposal: Class/Struct Initializer Method

### DISCUSSION THREAD

**Discussion by myrabyte** [dotnet/csharplang#6187] [Language Ideas]
Would be fantastic to have some kind of constructor or initializer method on a class or struct that runs AFTER all property initializers. So we can add logic with all the parameters already in place passed via init properties. This could be achieved by adding [init] attribute to the parameterless private constructor. Or add "init" keyword in front of it. Also, this method can be invoked by deserializers. Just a thought.

**Comment by svick** [comment]
This has been discussed before in https://github.com/dotnet/csharplang/discussions/3323 (as "validators") and https://github.com/dotnet/csharplang/discussions/3707 (as "final initializers"), though I didn't find a discussion just for this feature.

**Reply by Joe4evr** [reply]
@myrabyte Specifically, [this video with the LDT](https://youtu.be/fIXujTOA6BE?t=2749) indicates that this kind of feature is most definitely on their radar.

**Comment by GitClickOk** [comment]
an alternative is to use IAsyncLifetime as a first citizen interface like IEnumerable

**Comment by FaustVX** [comment]
This has been discussed in Final Initializer (#6591)
